ledger/participant-integration-api/src/main/scala/platform/store/backend/StorageBackend.scala

```diff
@@ -268,7 +268,9 @@ trait EventStorageBackend {
       transactionId: Ref.TransactionId,
       filterParams: FilterParams,
   )(connection: Connection): Vector[EventsTable.Entry[Raw.TreeEvent]]
-  def maxEventSeqIdForOffset(offset: Offset)(connection: Connection): Option[Long]
+
+  /** Max event sequential id of observable (create, consuming and nonconsuming exercise) events. */
+  def maxEventSequentialIdOfAnObservableEvent(offset: Offset)(connection: Connection): Option[Long]
```

Comment:
why this rename, and the restrictive enumeration of event types here?
(more importantly: why we seem to have a matching implementation for H2, but kept the original for other)

Reply:
The rename comes from the fact that this method does not include divulgence events sequential ids when computing the maximum value. In the previous version this was a bit hidden by using `participant_events` view. I think that the name `maxEventSeqIdForOffset()` is misleading in this context.

Reply:
To explain a bit more:
- Divulgence events have an `event_offset`
- In the view, the `event_offset` of divulgence events is hardcoded to `NULL` (not sure why, looks like a bug to me)
- Divulgence events are not observable over the ledger API. Only write side (command interpretation) queries should look at divulgence events.
- Including divulgence events in the `maxEventSeqIdForOffset` query significantly reduces its performance.

So excluding divulgence events for this query is the correct behavior, and we only get the correct behavior for Postgres/Oracle because of a bug in the view definition.

Reply:
I see. Can you please make TODO-s for Oracle and Postgres implementation to make this explicit?

Reply:
Sure, I'll add that.

Reply:
Done.